cpp/src/randomforest/randomforest_impl.cuh

```diff
@@ -257,20 +219,12 @@ void rfClassifier<T>::fit(const cumlHandle& user_handle, const T* input,
   for (int i = 0; i < this->rf_params.n_trees; i++) {
     int stream_id = omp_get_thread_num();
     unsigned int* rowids;
-    unsigned int* selected_ptr = nullptr;
-    char* temp_storage_ptr = nullptr;
-    if (this->rf_params.tree_params.split_algo != SPLIT_ALGO::GLOBAL_QUANTILE) {
-      rowids = sorted_selected_rows[stream_id]->data();
-      selected_ptr = sorted_selected_rows[stream_id]->data();
-      temp_storage_ptr = rows_temp_storage[stream_id]->data();
-    } else {
-      rowids = selected_rows[stream_id]->data();
-    }
-    this->prepare_fit_per_tree(
-      i, n_rows, n_sampled_rows, selected_rows[stream_id]->data(), selected_ptr,
-      temp_storage_ptr, temp_storage_bytes[stream_id],
-      tempmem[stream_id]->num_sms, local_handle[stream_id].getStream(),
-      local_handle[stream_id].getDeviceAllocator());
+    rowids = selected_rows[stream_id]->data();
+
+    this->prepare_fit_per_tree(i, n_rows, n_sampled_rows, rowids,
+                               tempmem[stream_id]->num_sms,
+                               local_handle[stream_id].getStream(),
```

Comment:
Forgive my forgetfulness, it's been awhile since we were discussing the use of local handles. Did we ever decide how to handle this so that we aren't maintaining an array of multiple handles? Are we waiting for #1015? 

If so, I'm okay integrating that change into a separate PR so that we can get this one merged.

Reply:
Yes that will be handled in PR #1015   
That is blocked due to this one. Once this get merged, we can make progress.